pkg/resources/view_acceptance_test.go

```diff
@@ -865,6 +864,51 @@ func TestAcc_View_Rename(t *testing.T) {
 	})
 }
 
+func TestAcc_View_Issue3073(t *testing.T) {
+	t.Setenv(string(testenvs.ConfigureClientOnce), "")
+	statement := "SELECT ROLE_NAME, ROLE_OWNER FROM INFORMATION_SCHEMA.APPLICABLE_ROLES"
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	viewModel := model.View("test", id.DatabaseName(), id.Name(), id.SchemaName(), statement)
+	viewModelWithColumns := model.View("test", id.DatabaseName(), id.Name(), id.SchemaName(), statement).WithColumnValue(config.SetVariable(
+		config.MapVariable(map[string]config.Variable{
+			"column_name": config.StringVariable("ROLE_NAME"),
+		}),
+		config.MapVariable(map[string]config.Variable{
+			"column_name": config.StringVariable("ROLE_OWNER"),
+		}),
+	))
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.View),
+		Steps: []resource.TestStep{
+			{
+				Config: accconfig.FromModel(t, viewModel),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_view.test", "name", id.Name()),
+				),
+			},
+			// specify the columns
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_View/basic"),
+				ConfigVariables: tfconfig.ConfigVariablesFromModel(t, viewModelWithColumns),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_view.test", plancheck.ResourceActionNoop),
+					},
+				},
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_view.test", "name", id.Name()),
+				),
```

Comment:
Could you add one step showing that going back to empty columns should also result in noop ?

Reply:
Done.